### ISSUE TITLE
TestHeatmap unit test failure with matplotlib 3.0.0

### ISSUE THREAD

**Issue by toddrme2178** [mwaskom/seaborn#1571]
I am running the unit tests for seaborn under matplotlib 3.0.0 and I am getting a failure.  This failure does not happen in the 2.x series of matplotlib.  I am not sure what the problem is.

```
=================================== FAILURES ===================================
_________________________ TestHeatmap.test_cbar_ticks __________________________

self = <seaborn.tests.test_matrix.TestHeatmap object at 0x7fbc03564828>

    def test_cbar_ticks(self):
        max_n_ticks = 3
    
        locator = mpl.ticker.MaxNLocator(max_n_ticks)
        f, (ax1, ax2) = plt.subplots(2)
        mat.heatmap(self.df_norm, ax=ax1, cbar_ax=ax2,
                    cbar_kws=dict(ticks=locator))
>       nt.assert_equal(len(ax2.yaxis.get_ticklabels()), max_n_ticks)

seaborn/tests/test_matrix.py:397: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib64/python3.6/unittest/case.py:829: in assertEqual
    assertion_func(first, second, msg=msg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <nose.tools.trivial.Dummy testMethod=nop>, first = 5, second = 3
msg = '5 != 3'

    def _baseAssertEqual(self, first, second, msg=None):
        """The default assertEqual implementation, not type specific."""
        if not first == second:
            standardMsg = '%s != %s' % _common_shorten_repr(first, second)
            msg = self._formatMessage(msg, standardMsg)
>           raise self.failureException(msg)
E           AssertionError: 5 != 3
```

It doesn't look like seaborn's travis currently tests matplotlib 3.0.0 so I am not sure if this is an error in my build or a general problem.

**Comment by JustinZhengBC** [comment]
My test fails even on matplotlib 2.0.0 and 2.1.2, and many recent PR's are failing because of this test too.

Printing the ticks given in the test gives
```
Text(1, -2.0, '−2')
Text(1, -1.0, '−1')
Text(1, 0.0, '0')
Text(1, 1.0, '1')
Text(1, 2.0, '2')
```
so although in this case the ticker returns the correct number of ticks (-2 and 2 are not visible) calling `len` on this returns 5 while only 3 are visible.

**Comment by mwaskom** [comment]
Thanks, fixed in #1712
